weather_forecast_retrieval/hrrr_preprocessor.py

```diff
@@ -120,12 +157,10 @@ def run(self):
             new_hrrr_file = os.path.join(new_hrrr_path, hrrr_file_name)
 
             # create a temp file with just the variables needed
-            variable_action = """wgrib2 {} -match """ \
-                """'TMP:2 m|RH:2 m|UGRD:10 m|VGRD:10 m|APCP:surface|""" \
-                """DSWRF:surface|HGT:surface|TCDC:entire atmosphere' -GRIB {}"""
-
-            variable_action = variable_action.format(hrrr_abs_file_path,
-                                                     self.tmp_file)
+            variable_action = "wgrib2 {} -match '{}' -GRIB {}".format(
+                hrrr_abs_file_path,
+                self.variable_match,
+                self.tmp_file)
```

Comment:
I would pipe the two commands instead of creating a temp file. This will prevent unnecessary disk I/O.
I have done it this way: https://github.com/UofU-Cryosphere/isnoda/blob/master/scripts/download_hrrr.sh

Reply:
Never tried wgrib2 with piping, good thing it works!

Reply:
I also would have not thought of using them if I would not have found this here: https://www.cpc.ncep.noaa.gov/products/wesley/wgrib2/pipes.html

Reply:
How do you control the number of threads?
My approach with `wgrib2` is to set the `-ncpu` option

Reply:
There is a ncpu option if you set it via the cli, defaults to not setting it. Although in my testing, it doesn't seem to work with wgrib v2.0.8